MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionViewQtGUI.cpp

```diff
@@ -387,19 +387,81 @@ std::string EnggDiffractionViewQtGUI::getRBNumber() const {
 }
 
 std::string EnggDiffractionViewQtGUI::currentVanadiumNo() const {
-  return m_uiTabCalib.lineEdit_current_vanadium_num->text().toStdString();
+  std::string vanstr =
+      m_uiTabCalib.lineEdit_current_vanadium_num->text().toStdString();
+  if (vanstr.length() > 6) {
+    std::string vanno = "";
```

Comment:
Aha I see the "6" here.

This will work in practice. But to make it bullet-proof and neater I'd suggest:

- Replacing the check `>6` by a more specific check: can it be converted to a integer? (with boost::lexical_cast, which will throw an exception if it cannot do the cast).
- Notice that you do the same check in 7 places. The check should go into a method which you can call `isValidRunNumber` or similar.
- Notice also that the check on the filename and loop over the digits is also the same in those 7 places, so it should also go to a small method.
- When you do the check like that we are actually making more resilient, and we can give an specific exception+error message telling the user that this is not a valid run number.

An example case where the current check would fail: I'm a lousy user and type `003af`. You'll anyway get an exception error later when Load tries to find that funny run, but it would be better if we can prevent that (we cannot rely on users really paying attention to the red symbol next to the MWRun widget).